File: DSA/VenikaPanwar_590013755/Day_78_Question_3.java
public class Day_78_Question_3 {

    static class TreeNode {
        int val;
        TreeNode left, right;
        TreeNode(int val) { this.val = val; }
    }

    // Finds minimum value in BST
    public static int findMin(TreeNode root) {
        if (root == null) throw new IllegalArgumentException("Tree is empty");
        while (root.left != null) root = root.left;
        return root.val;
    }

    // Finds maximum value in BST
    public static int findMax(TreeNode root) {
        if (root == null) throw new IllegalArgumentException("Tree is empty");
        while (root.right != null) root = root.right;
        return root.val;
    }

    public static void main(String[] args) {
        // Example: Tree 8 / \ 3 10 / \ \ 1 6 14
        TreeNode root = new TreeNode(8);
        root.left = new TreeNode(3);
        root.right = new TreeNode(10);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(6);
        root.right.right = new TreeNode(14);

        System.out.println("Min: " + findMin(root) + ", Max: " + findMax(root)); // Output: Min = 1, Max = 14

        // Example: Tree 5
        TreeNode root2 = new TreeNode(5);
        System.out.println("Min: " + findMin(root2) + ", Max: " + findMax(root2)); // Output: Min = 5, Max = 5
    }
}

File: DSA/SuryanshMishra_590012069/Day78/day78question3.cpp
#include <cstddef>
#include <climits>
using namespace std;

// Definition for a binary search tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

// Find minimum value in a BST (leftmost node). Returns INT_MAX for empty tree.
int findMinBST(TreeNode* root) {
    if (root == nullptr) return INT_MAX;
    TreeNode* cur = root;
    while (cur->left) cur = cur->left;
    return cur->val;
}

// Find maximum value in a BST (rightmost node). Returns INT_MIN for empty tree.
int findMaxBST(TreeNode* root) {
    if (root == nullptr) return INT_MIN;
    TreeNode* cur = root;
    while (cur->right) cur = cur->right;
    return cur->val;
}
